### ISSUE TITLE
Request interception is not working

### ISSUE THREAD

**Issue by mockfox** [atroche/rust-headless-chrome#280]
I enable request interception with tab.enable_request_interception(), but network events are not being intercepted. Am I missing something?

**Comment by mdrokz** [comment]
> I enable request interception with tab.enable_request_interception(), but network events are not being intercepted. Am I missing something?

Hi, can you post the code and version of rust-headless-chrome you are using?

**Comment by mockfox** [comment]
This is the code I am using:
```
    tab.enable_request_interception(Arc::new(|_transport, _session_id, _params| {
        println!("request!");
        headless_chrome::browser::tab::RequestPausedDecision::Continue(None)
    }))?;
    
    tab.navigate_to(url)?;
    tab.wait_until_navigated()?;
```
The interception closure is not called at any point.

**Comment by mdrokz** [comment]
> This is the code I am using:
> 
> ```
>     tab.enable_request_interception(Arc::new(|_transport, _session_id, _params| {
>         println!("request!");
>         headless_chrome::browser::tab::RequestPausedDecision::Continue(None)
>     }))?;
>     
>     tab.navigate_to(url)?;
>     tab.wait_until_navigated()?;
> ```
> 
> The interception closure is not called at any point.

Which version of the library are you using and whats the url  ?

**Comment by mockfox** [comment]
I am using the latest version from the git repository since the one on crates.io does not support launch arguments. It worked on crates.io version 0.9.0 (IIRC) when `enable_request_interception()` required a `RequestPattern`.

**Comment by mockfox** [comment]
It seems like `RequestInterceptor` should be using `RequestInterceptedEvent` rather than `RequestPausedEvent`.

**Comment by mockfox** [comment]
It is also not clear to me how to use a `RequestPattern` like in the crates.io version to only intercept certain requests.

**Comment by zhan3333** [comment]
@mockfox this test is working for me:

https://github.com/atroche/rust-headless-chrome/blob/61ce783806e5d75a03f731330edae6156bb0a2e0/tests/simple.rs#L592-L630

**Comment by mockfox** [comment]
> @mockfox this test is working for me:
> 
> https://github.com/atroche/rust-headless-chrome/blob/61ce783806e5d75a03f731330edae6156bb0a2e0/tests/simple.rs#L592-L630

I haven't had a chance to test it myself, but I was having issues intercepting requests originating from HTML/JS rather than from this crate.

**Comment by zhan3333** [comment]
> 

I didn't understand your question, can you attach the usage code?

**Comment by zhan3333** [comment]
> > @mockfox this test is working for me:
> > https://github.com/atroche/rust-headless-chrome/blob/61ce783806e5d75a03f731330edae6156bb0a2e0/tests/simple.rs#L592-L630
> 
> I haven't had a chance to test it myself, but I was having issues intercepting requests originating from HTML/JS rather than from this crate.

I didn't understand your question, can you attach the usage code?

**Comment by escritorio-gustavo** [comment]
I have the same issue

Cargo.toml:
```toml
headless_chrome = { git = "https://github.com/atroche/rust-headless-chrome", features = [
  "fetch",
] }
```

Code:
```rust
tab.enable_request_interception(Arc::new(move |_, _, event: RequestPausedEvent| {
    let url = event.params.request.url;

    // Doesn't print anything
    println!("{url}");

    // No request was blocked
    if url.ends_with(".css")
        || url.ends_with(".woff2")
        || url.ends_with(".svg")
        || url.ends_with(".png")
        || url.ends_with(".ico")
    {
        RequestPausedDecision::Fail(FailRequest {
            request_id: event.params.request_id,
            error_reason: ErrorReason::Aborted,
        })
    } else {
        RequestPausedDecision::Continue(None)
    }
}))?;
```

**Comment by Bowserinator** [comment]
Did you add the request patterns like in the example? That fixed it for me

**Comment by escritorio-gustavo** [comment]
Yes, adding the patterns like the link @zhan3333 provided worked for me, even allowing me to drop the tracing_subscriber call that is in the examples directory
